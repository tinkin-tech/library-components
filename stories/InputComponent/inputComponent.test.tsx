import * as renderer from 'react-test-renderer'
import { render, fireEvent, getByTestId } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'

import {
  inputTypeText,
  inputTypeNumber,
  inputDefault,
  InputTypeTextWithValue,
  InputWithError,
} from './fixture/input.fixture'

describe('InputComponent test', () => {
  it('Should take the snapshot with require props', () => {
    const input = renderer.create(inputDefault)
    const inputComponentJSON = input.toJSON()
    expect(inputComponentJSON).toMatchSnapshot()
  })

  it('Should render input in virtual DOM', () => {
    const { getByTestId } = render(inputTypeText)
    expect(getByTestId('input-component')).toBeInTheDocument()
  })

  it('Should be able to enter a value, when type is text', () => {
    const dom = render(inputTypeText)
    const input = getByTestId(dom.container, 'input-component')
    fireEvent.change(input, { target: { value: 'Hello World!' } })
    expect(input.getAttribute('value')).toBe('Hello World!')
  })

  it('The value should be empty when I enter text and type is number', () => {
    const dom = render(inputTypeNumber)
    const input = getByTestId(dom.container, 'input-component')
    fireEvent.change(input, { target: { value: 'Hello World!' } })
    expect(input.getAttribute('value')).toBe('')
  })

  it('Should pass value different from valueState of input component', () => {
    const { getByTestId, rerender } = render(inputTypeText)
    expect(getByTestId('input-component')).toMatchObject({ value: '' })
    rerender(InputTypeTextWithValue)
    expect(getByTestId('input-component')).toMatchObject({
      value: InputTypeTextWithValue.props.value,
    })
  })

  it('Should add className label-error when component pass with error', () => {
    const { getByTestId, rerender } = render(inputTypeText)
    expect(getByTestId('label-component').className).not.toContain(
      'label-error'
    )
    rerender(InputWithError)
    expect(getByTestId('label-component').className).toContain('label-error')
  })
})
