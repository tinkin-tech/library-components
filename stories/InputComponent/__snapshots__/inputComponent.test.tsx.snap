// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputComponent test Should take the snapshot with require props 1`] = `
<div
  className="input-component input flex-column "
  id="input-component"
>
  <label
    htmlFor="input"
  >
    <label
      className="label "
      htmlFor="input"
    >
      Label*
    </label>
  </label>
  <label
    className="label "
    htmlFor="input"
  >
    Label*
  </label>
  <input
    autoComplete="off"
    className=""
    data-testid="input-component"
    disabled={false}
    id="input"
    name="input"
    onChange={[Function]}
    placeholder="placeholder"
    spellCheck={false}
    type="text"
    value="value"
  />
  
  
</div>
`;
